File: library/src/main/java/com/uploadcare/android/library/api/PaginatedQueryBuilder.java
package com.uploadcare.android.library.api;

import java.util.List;

public interface PaginatedQueryBuilder<T> {

    /**
     * Returns a resource iterable for lazy loading.
     */
    Iterable<T> asIterable();

    /**
     * Iterates through all resources and returns a complete list.
     */
    List<T> asList();

}

File: library/src/main/java/com/uploadcare/android/library/api/PaginatedQueryBuilder.kt
package com.uploadcare.android.library.api

interface PaginatedQueryBuilder<T> {

    /**
     * Returns a resource iterable for lazy loading.
     */
    fun asIterable(): Iterable<T>

    /**
     * Iterates through all resources and returns a complete list.
     */
    fun asList(): List<T>
}